charts/filecoin-chain-archiver/templates/reset-cronjob.yaml

```diff
@@ -3,13 +3,13 @@
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  name: {{ printf "%s-ds-reset-%s" $.Release.Name $reset.pod | trunc -52 | trimPrefix "-" }}
+  name: {{ include "filecoin-chain-archiver.podResets.name" (dict "i" $ "reset" $reset) }}
```

Comment:
Do you need to pass the global pointer in? Isn't it accessible by default?

Reply:
No, because `include` only accepts one argument.
To pass the global pointer, one would have to do 
```
{{ include "filecoin-chain-archiver.podResets.name" $ }}
```
Which then satisfies a reference like `{{ .Values.foo }}`, but the $reset variable isn't in this pointer, so this error ensues
```
Error: template: filecoin-chain-archiver/templates/reset-cronjob.yaml:6:11: executing "filecoin-chain-archiver/templates/reset-cronjob.yaml" at <include "filecoin-chain-archiver.podResets.name" $>: error calling include: template: filecoin-chain-archiver/templates/_helpers.tpl:97:48: executing "filecoin-chain-archiver.podResets.name" at <.reset.pod>: nil pointer evaluating interface {}.pod
```
To pass the global pointer and  `$reset`  struct, the only way is to assign both to a `dict`

I agree it seems janky, and if there is actually a better way of doing this, I'm very open to changing it in another iteratoin

Reply:
You're right, I guess I was mis-remembering how the scoping works. This [comment ](https://github.com/helm/helm/issues/7634#issuecomment-851057501) is interesting, though I think your approach is safer